src/psyclone/tests/psyGen_test.py

```diff
@@ -2617,8 +2627,269 @@ def test_binaryoperation_can_be_printed():
+
+    # Declare a duplicate name symbol
+    with pytest.raises(InternalError) as error:
+        sym_table.declare("var1", "real", 0, "write_arg")
+    assert "Multiple definition of symbol: 'var1'." in str(error.value)
+
+
+def test_symboltable_lookup():
+    sym_table = SymbolTable()
+    sym_table.declare("var1", "real", 3, "read_arg")
+    sym_table.declare("var2", "integer", 0, "write_arg")
+    sym_table.declare("var3", "real", 0, None)
+
+    assert isinstance(sym_table.lookup("var1"), Symbol)
+    assert sym_table.lookup("var1").name == "var1"
+    assert isinstance(sym_table.lookup("var2"), Symbol)
+    assert sym_table.lookup("var2").name == "var2"
+    assert isinstance(sym_table.lookup("var3"), Symbol)
+    assert sym_table.lookup("var3").name == "var3"
+
+    with pytest.raises(KeyError) as error:
+        sym_table.lookup("notdeclared")
+
+
+def test_symboltable_view(capsys):
+    sym_table = SymbolTable()
+    sym_table.declare("var1", "real", 3, "read_arg")
+    sym_table.declare("var2", "integer", 2, "read_arg")
+    sym_table.view()
+    output, _ = capsys.readouterr()
+    assert "Symbol Table:\n" in output
+    assert "var1" in output
+    assert "var2" in output
+
+
+def test_symboltable_can_be_printed():
+    sym_table = SymbolTable()
+    sym_table.declare("var1", "real", 3, "read_arg")
+    sym_table.declare("var2", "integer", 2, "read_arg")
+    assert "Symbol Table:\n" in str(sym_table)
+    assert "var1" in str(sym_table)
+    assert "var2" in str(sym_table)
+
+
 # Test Fparser2ASTProcessor
 
+def test_fparser2astprocessor_generate_schedule():
+    ''' Tests the fparser2AST generate_schedule method.
+    '''
+    ast1 = fpapi.parse(FAKE_KERNEL_METADATA, ignore_comments=True)
+    metadata = DynKernMetadata(ast1)
+    my_kern = DynKern()
+    my_kern.load_meta(metadata)
+    ast2 = my_kern.ast
+    processor = Fparser2ASTProcessor()
+
+    # Test properly formed but empty kernel module
+    schedule = processor.generate_schedule("dummy_code", ast2)
+    assert isinstance(schedule, KernelSchedule)
+
+    # Test generate an unexistent subroutine name
+    with pytest.raises(InternalError) as error:
+        schedule = processor.generate_schedule("nonexistent_code", ast2)
+    assert "Unexpected kernel AST. Could not find " \
+           "subroutine: nonexistent_code" in str(error.value)
+
+    # Test corrupting ast by deleting subroutine
+    del ast2.content[0].content[2]
+    with pytest.raises(InternalError) as error:
+        schedule = processor.generate_schedule("dummy_code", ast2)
+    assert "Unexpected kernel AST. Could not find " \
+           "subroutine: dummy_code" in str(error.value)
+
+    # Test corrupting ast by deleting specification part
+    del ast2.content[0].content[1]
+    with pytest.raises(InternalError) as error:
+        schedule = processor.generate_schedule("dummy_code", ast2)
+    assert "Unexpected kernel AST. Could not find specification part."
+
+    DUMMY_KERNEL_METADATA = '''
```

Comment:
I think you should break the rest of this test out into a separate one as it deals with a non-empty kernel.

Reply:
Done